File: LovePlay-RAC/LovePlay/Class/Zone/View/DiscussListHeaderView.m
//
//  DiscussListHeaderView.m
//  LovePlay
//
//  Created by Yuns on 2017/2/3.
//  Copyright © 2017年 Yuns. All rights reserved.
//

#import "DiscussListHeaderView.h"
#import "DiscussImageModel.h"

@interface DiscussListHeaderView ()
//UI
@property (nonatomic, strong) UIImageView *imageNode;
@property (nonatomic, strong) UILabel *titleTextLabel;
@property (nonatomic, strong) UILabel *descriptionTextLabel;
//Data
@property (nonatomic, strong) DiscussImageModel *imageModel;
@end

@implementation DiscussListHeaderView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self addSubnodes];
        [self mas_subViews];
    }
    return self;
}

- (void)addSubnodes
{
    [self addSubview:self.imageNode];
    
    [self addSubview:self.titleTextLabel];
    
    [self addSubview:self.descriptionTextLabel];
}

- (void)mas_subViews
{
    [_imageNode mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(self);
    }];
    
    [_descriptionTextLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self).offset(10);
        make.bottom.equalTo(self).offset(-10);
        make.right.equalTo(self).offset(-10);
    }];
    
    [_titleTextLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.equalTo(_descriptionTextLabel);
        make.bottom.equalTo(_descriptionTextLabel.mas_top).offset(-5);
    }];
}

#pragma mark - public
- (void)setupImageModel:(DiscussImageModel *)imageModel
{
    _imageModel = imageModel;
    _imageNode.imageURL = [NSURL URLWithString:imageModel.bannerUrl];
    _titleTextLabel.text = imageModel.modelName;
    _descriptionTextLabel.text = [NSString stringWithFormat:@"今日：%@",imageModel.todayPosts];
}

#pragma mark - setter / getter
- (UIImageView *)imageNode
{
    if (!_imageNode) {
        UIImageView *imageNode = [[UIImageView alloc] init];
        imageNode.contentMode = UIViewContentModeScaleAspectFill;
        imageNode.clipsToBounds = YES;
        _imageNode = imageNode;
    }
    return _imageNode;
}


- (UILabel *)titleTextLabel
{
    if (!_titleTextLabel) {
        UILabel *titleTextLabel = [[UILabel alloc] init];
        titleTextLabel.font = [UIFont systemFontOfSize:18];
        titleTextLabel.textColor = [UIColor whiteColor];
        _titleTextLabel = titleTextLabel;
    }
    return _titleTextLabel;
}

- (UILabel *)descriptionTextLabel
{
    if (!_descriptionTextLabel) {
        UILabel *descriptionTextLabel = [[UILabel alloc] init];
        descriptionTextLabel.font = [UIFont systemFontOfSize:12];
        descriptionTextLabel.textColor = [UIColor whiteColor];
        _descriptionTextLabel = descriptionTextLabel;
    }
    return _descriptionTextLabel;
}


@end

File: LovePlay-Swift3/LovePlay/Class/Zone/View/DiscussListHeaderView.swift
//
//  DiscussListHeaderView.swift
//  LovePlay
//
//  Created by weiying on 2017/6/6.
//  Copyright © 2017年 yuns. All rights reserved.
//

import UIKit

class DiscussListHeaderView: UIView {

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.addSubViews()
        self.snp_subViews()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    //MARK: - private
    func addSubViews() {
        self.addSubview(self.imgView)
        self.addSubview(self.titleTextLabel)
        self.addSubview(self.descTextLabel)
    }
    
    func snp_subViews() {
        self.imgView.snp.makeConstraints { (make) in
            make.edges.equalToSuperview();
        }
        
        self.titleTextLabel.snp.makeConstraints { (make) in
            make.left.right.equalTo(self.descTextLabel);
            make.bottom.equalTo(self.descTextLabel.snp.top).offset(-5);
        }
        
        self.descTextLabel.snp.makeConstraints { (make) in
            make.left.equalToSuperview().offset(10);
            make.bottom.equalToSuperview().offset(-10);
            make.right.equalToSuperview().offset(-10);
        }
    }
    
    fileprivate var _imgModel : DiscussImgModel?
    var imgModel : DiscussImgModel? {
        set {
            _imgModel = newValue
            let imgURL = URL(string: (newValue?.bannerUrl)!)
            self.imgView.kf.setImage(with: imgURL)
            
            self.titleTextLabel.text = newValue?.modelName
            self.descTextLabel.text = "今日：" + "\((newValue?.todayPosts)!)"
        }
        
        get {
            return _imgModel
        }
    }
    
    //MARK: - setter / getter
    lazy var imgView : UIImageView = {
        let imgView : UIImageView = UIImageView()
        imgView.clipsToBounds = true
        return imgView
    }()
    
    lazy var titleTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.textColor = UIColor.white
        label.font = UIFont.systemFont(ofSize: 18)
        return label
    }()
    
    lazy var descTextLabel : UILabel = {
        let label : UILabel = UILabel()
        label.textColor = UIColor.white
        label.font = UIFont.systemFont(ofSize: 12)
        return label
    }()
}
